src/mapblock.h

```diff
@@ -31,6 +31,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "nodetimer.h"
 #include "modifiedstate.h"
 #include "util/numeric.h" // getContainerPos
+#include "settings.h"
```

Comment:
Afaik, the settings get only does it once (although I'm not sure whether
inline affects the static at all) it should be const though.
On 21 Jul 2015 07:18, "Zeno-" <notifications@github.com> wrote:

> In src/mapblock.h
> <https://github.com/minetest/minetest/pull/2908#discussion_r35072623>:
>
> > @@ -31,6 +31,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
> >  #include "nodetimer.h"
> >  #include "modifiedstate.h"
> >  #include "util/numeric.h" // getContainerPos
> > +#include "settings.h"
>
> There is probably no reason for the function to be inline anyway because
> the majority of the processor time is going to be spent getting the value
> of the setting (by far) (i.e. line 634). You could change static u16
> map_gen_limit to const static u16 map_gen_limit I guess and then it might
> make sense for the function to be inline.
>
> Actually, the bigger problem is that the inline function doesn't seem to
> be used in places where it should be...
>
> Just by scrolling up in this PR I can see:
> src/clientiface.cpp lines 225-
> src/map.cpp lines 2656
>
> ... that appear to do exactly what the function does
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/minetest/minetest/pull/2908/files#r35072623>.
>